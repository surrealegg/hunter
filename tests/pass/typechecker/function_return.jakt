extern function printf(anon format: raw c_char) -> usize

function case_1() {
    printf("CASE 1: Implicit return void\n")
    return
}

function case_2() -> void {
    printf("CASE 2: Explicit return void\n")
    return
}

function case_3() -> bool {
    printf("CASE 3: Explicit return bool\n")
    return true
}

function case_4() {
    {
        printf("CASE 4: Implicit return bool block\n")
        return true
    }
}

function case_5() {
    printf("CASE 4: Implicit return i64 if\n")
    if false {
        return 100
    }
    return 42
}

function case_6() {
    printf("CASE 4: Implicit return i64 if/else\n")
    if false {
        return 100
    } else {
        return 42
    }
}

function typeck() {
    case_1()
    case_2()
    case_3()
    case_4()
    case_5()
    case_6()
}

function main() {
    return typeck();
}