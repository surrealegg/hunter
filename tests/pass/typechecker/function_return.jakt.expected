CASE 1: Implicit return void
CASE 2: Explicit return void
CASE 3: Explicit return bool
CASE 4: Implicit return bool block
CASE 4: Implicit return i64 if
CASE 4: Implicit return i64 if/else
